src/core/thread/network_data_local.cpp

```diff
@@ -69,6 +68,47 @@ Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
     // of an IEEE 802.15.4 interface MUST have a length of 64 bits.
     VerifyOrExit(!aConfig.mSlaac || aConfig.mPrefix.mLength == OT_IP6_PREFIX_BITSIZE, error = kErrorInvalidArgs);
 
+    // Validate Prefix
+    VerifyOrExit((aConfig.GetPrefix().GetLength() > 0) && aConfig.GetPrefix().IsValid(), error = kErrorInvalidArgs);
+
+    // Validate Preference
+    switch (aConfig.mPreference)
+    {
+    case OT_ROUTE_PREFERENCE_LOW:
+    case OT_ROUTE_PREFERENCE_MED:
+    case OT_ROUTE_PREFERENCE_HIGH:
+        break;
+    default:
+        ExitNow(error = kErrorInvalidArgs);
+    }
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if (aConfig.mDp)
+    {
+        // Check if domain prefix is already configured
+        OnMeshPrefixConfig prefixConfig;
+        VerifyOrExit(Get<BackboneRouter::Local>().GetDomainPrefix(prefixConfig) == kErrorNotFound,
```

Comment:
Do we need to `CheckBufferSizeForPrefix` for the domain prefix too?

Reply:
There can be only one domain prefix set to mDomainPrefixConfig in bbr_local. I think we need to check only if there was a previous one configured. The question is if we need to check if domain prefix is already configured to prohibit overwritting or allow to set domain prefix without removing the previous entry ?

Reply:
`NetworkData::Local::AddPrefix` always calls `RemovePrefix` to remove the same prefix from Network Data before adding it. So, I think we should keep this behavior for Domain Prefix unless there is a good reason not to. 

Thoughts? @XPAWAN @abtink @librasungirl

Reply:
> There can be only one domain prefix set to mDomainPrefixConfig in bbr_local. I think we need to check only if there was a previous one configured. The question is if we need to check if domain prefix is already configured to prohibit overwritting or allow to set domain prefix without removing the previous entry ?

See above comment, domain prefix is designed to allow in-place change, without requiring clearing previous config.